Model/Checkout/Payment/MonduPlaceOrderService.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mondu\MonduPaymentHyva\Model\Checkout\Payment;
+
+use Exception;
+use Hyva\Checkout\Model\Magewire\Component\EvaluationResultFactory;
+use Hyva\Checkout\Model\Magewire\Component\EvaluationResultInterface;
+use Hyva\Checkout\Model\Magewire\Payment\AbstractPlaceOrderService;
+use Hyva\Checkout\Model\Magewire\Payment\DefaultOrderData;
+use Hyva\Checkout\Model\Magewire\Payment\PlaceOrderServiceInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Quote\Api\CartManagementInterface;
+use Magento\Quote\Model\Quote;
+use Magewirephp\Magewire\Component;
+
+class MonduPlaceOrderService extends AbstractPlaceOrderService implements PlaceOrderServiceInterface
+{
+    public function __construct(
+        private readonly CustomerDataProvider $customerDataProvider,
+        private readonly SessionStorage $sessionStorage,
+        private readonly TransactionProcessor $transactionProcessor,
+        CartManagementInterface $cartManagement,
+        ?DefaultOrderData $orderData = null
+    ) {
+        parent::__construct($cartManagement, $orderData);
+    }
+
+    public function placeOrder(Quote $quote): int
+    {
+        try {
+            $response = $this->transactionProcessor->process([
+                'email' => $this->customerDataProvider->getEmail($quote),
+                'user-agent' => $this->customerDataProvider->getUserAgent(),
+                'payment_method' => $this->customerDataProvider->getPaymentMethod($quote),
+            ]);
+
+            if (empty($response['token']) || !empty($response['error'])) {
+                throw new LocalizedException(
+                    __('Mondu API error: %1', $response['message'] ?? 'Invalid response')
+                );
+            }
+            $this->sessionStorage->saveResponse($response);
+
+            return 0;
+        } catch (\Throwable $e) {
+            throw new LocalizedException(__('Failed to initiate Mondu order: %1', $e->getMessage()));
+        }
+    }
+
+    public function evaluateCompletion(EvaluationResultFactory $resultFactory, ?int $orderId = null): EvaluationResultInterface
+    {
+        $response = $this->sessionStorage->getResponse();
+        $this->sessionStorage->clearResponse();
+
+        if ($response && $response['source'] === 'hosted' && !empty($response['hosted_checkout_url'])) {
+            return $resultFactory->createRedirect($response['hosted_checkout_url']);
+        }
+
+        if ($response && $response['source'] === 'widget') {
```

Comment:
non-blocking: @ivan-pugach-mondu  we don't need to support widget anymore for the newer versions.

You can keep it here since you already implemented but we are sun setting widget and pushing existing merchants to switch to HC